crates/next-api/src/app.rs

```diff
@@ -862,7 +862,41 @@ impl AppEndpoint {
                 }
                 let client_shared_availability_info = client_shared_chunk_group.availability_info;
 
-                let client_references = client_reference_graph(Vc::cell(vec![rsc_entry_asset]));
+                let client_references = {
+                    let modules = if let Some(module_part_asset) =
+                        Vc::try_resolve_downcast_type::<EcmascriptModulePartAsset>(rsc_entry)
+                            .await?
```

Comment:
Casting is not good. It will break when e. g. tree shaking is not enabled.

Reply:
Is there some other way to figure out the layout segments? For all the other functions later on, we use the result of this very function. But here we need to know this even beforehand.

Maybe a graph walking of just 1 or a few levels?